### PR TITLE
API: Add ChangePassword implementation

### PR DESCRIPTION
**Pull request by gchaincl**

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -203,34 +203,34 @@ func (acc *Account) loginChallenge(username string) (*LoginChallengeResponse, er
...
 	if err != nil {
 		return err
 	}
 
-	buf, err := Marshal(&LoginRequest{
+	req := LoginRequest{
 		Username:  username,
-		Challenge: challenge.Challenge,
+		Challenge: lc.Challenge,
 		Host:      "api.etebase.com",
```
Comment (tasn) [inline client.go:226]: Not good to hardcode it. It should be passed as the last parameter.

File: client.go [hunk 2]
```diff
@@ -241,22 +241,79 @@ func (acc *Account) Login(username, password string) error {
 	}
 	defer resp.Body.Close()
 
+	if resp.StatusCode == http.StatusUnauthorized {
```
Comment (tasn) [inline client.go:244]: What we did in the other clients is have the status code handling all in one place. So we just map status to error codes and decode the failures. Is this something that's applicable to Go too? So we just automatically handle all of the error types centrally?

File: client.go [hunk 3]
```diff
@@ -241,22 +241,79 @@ func (acc *Account) Login(username, password string) error {
...
+
+	encrypedContent, err := crypto.Encrypt(acc.mainKey, append(acc.accountKey, acc.idPriv...))
+	if err != nil {
+		return err
+	}
+
+	req := LoginRequest{
+		Username:         acc.session.User.Username,
+		Challenge:        lc.Challenge,
+		Host:             "api.etebase.com",
```
Comment (tasn) [inline client.go:282]: Same about hardcoding the url. Should be saved in the instance.

Comment (gchaincl) [reply client.go:282]: agreed!

File: client.go (hunk 2 above)
Comment (gchaincl) [reply client.go:244]: we should. I will address this on a different pr.

**Pull request merged**